flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputSelectableProcessor.java

```diff
@@ -280,11 +286,17 @@ else if (recordOrMark.isWatermark()) {
 		}
 	}
 
+	private void updateInputSelection() {
+		if (inputSelector != null) {
+			inputSelection = inputSelector.nextSelection();
+		}
```

Comment:
maybe add an else branch with 

```
else {
  inputSelection == InputSelection.ALL;
}
```
and re-use this in `prepareForProcessing`?

I don't think that would any measurable performance overhead, while would reduce code duplication by a bit?